### ISSUE TITLE
Chuffed broken on ubuntu 16.04 LTS?

### ISSUE THREAD

**Issue by matsc-at-sics-se** [MiniZinc/libminizinc#230]
Did I do something wrong during installation?
[dna.mzn.zip](https://github.com/MiniZinc/libminizinc/files/2323836/dna.mzn.zip)
```
matsc@teterev:~/MZN$ minizinc -v --solver chuffed --fzn dna.fzn -D N=20 dna.mzn
MiniZinc to FlatZinc converter, version 2.2.0, build 28502108
Copyright (C) 2014-2018 Monash University, NICTA, Data61
Parsing file(s) 'dna.mzn', 'cmd:/N=20' ...
[...]
Printing FlatZinc to 'dna.fzn' ... done (0.54 s)
Maximum memory 8 Mbytes.
  Flattening done, 0.54 s
      % SOLVING PHASE
FZN solver plugin, compiled  Aug 24 2018  00:38:11
Using FZN solver /opt/MiniZincIDE-2.2.0-bundle-linux/bin/fzn-chuffed for solving, parameters: -v 
=====ERROR=====
   Done (overall time 0.68 s).
```
```
matsc@teterev:~/MZN$ minizinc -v --solver chuffed dna.fzn
MiniZinc to FlatZinc converter, version 2.2.0, build 28502108
Copyright (C) 2014-2018 Monash University, NICTA, Data61
Parsing file(s) 'dna.fzn' ...
[...]
MiniZinc: type error: no function or predicate with this signature found: `table_int(array[int] of var int,array[int] of int)'
   Done (overall time 0.26 s).
```

**Comment by a1880** [comment]
I've tried your command with MiniZinc 2.2.0 under Windows. It works without problems on my PC.
When I start the same out of the IDE, I do not encounter any problems as well.
So, it is probably something special with the Ubuntu port/environment.

The error message complaints about table_int() with second argument as 1D array, while the libraries have a 2D array as second parameter. I have no clue, why this is happening.

The output log looks like follows:

C:\ak\programme\MiniZinc IDE (bundled) 2.2.0>minizinc -v --solver chuffed --fzn dna.fzn -D N=20 dna.mzn
MiniZinc to FlatZinc converter, version 2.2.0, build 28502108
Copyright (C) 2014-2018 Monash University, NICTA, Data61
Parsing file(s) 'dna.mzn', 'cmd:/N=20' ...
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/stdlib.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/builtins.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/redefinitions-2.1.1.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/redefinitions-2.1.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/redefinitions-2.0.2.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/redefinitions-2.0.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/chuffed/redefinitions.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/flatzinc_builtins.mzn'
processing file 'dna.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/globals.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/value_precede_chain.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/value_precede_chain_set.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/value_precede_chain_int.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/value_precede.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/value_precede_set.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/value_precede_int.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/table.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/chuffed/table_int.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/table_bool.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/symmetric_all_different.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/sum_pred.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/chuffed/subcircuit.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/strict_lex2.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/span.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/sort_fn.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/sort.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/alldifferent.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/sliding_sum.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/roots_fn.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/roots.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/regular_regexp.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/regular_nfa.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/chuffed/regular.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/range_fn.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/range.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/partition_set.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/nvalue_fn.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/chuffed/nvalue.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/network_flow.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/minimum.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/member.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/set_member.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/member_set.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/member_int.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/member_float.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/member_bool.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/maximum.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/link_set_to_booleans.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/lex2.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/lex_less.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/lex_less_set.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/lex_less_int.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/lex_less_float.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/lex_less_bool.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/lex_lesseq.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/lex_lesseq_set.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/lex_lesseq_int.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/lex_lesseq_float.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/lex_lesseq_bool.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/lex_greatereq.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/lex_greater.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/knapsack.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/inverse_set.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/inverse_fn.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/chuffed/inverse.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/int_set_channel.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/increasing.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/increasing_set.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/increasing_int.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/increasing_float.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/increasing_bool.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/global_cardinality_low_up_closed.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/chuffed/global_cardinality_low_up.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/global_cardinality_closed_fn.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/global_cardinality_closed.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/global_cardinality_fn.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/global_cardinality.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/geost.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/exactly.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/exactly_set.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/chuffed/exactly_int.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/element.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/element_set.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/element_int.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/element_float.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/element_bool.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/distribute_fn.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/chuffed/distribute.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/disjunctive_strict_opt.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/disjunctive_opt.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/chuffed/disjunctive_strict.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/chuffed/disjunctive.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/disjoint.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/diffn_nonstrict_k.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/diffn_k.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/diffn_nonstrict.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/diffn.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/decreasing.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/decreasing_set.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/decreasing_int.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/decreasing_float.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/decreasing_bool.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/cumulative_opt.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/chuffed/cumulative.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/count_lt.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/count_leq.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/count_gt.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/count_geq.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/count_neq.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/count_eq.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/count_fn.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/chuffed/count.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/chuffed/circuit.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/bin_packing_load_fn.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/bin_packing_load.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/bin_packing_capa.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/bin_packing.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/at_most1.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/at_most.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/at_most_set.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/chuffed/at_most_int.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/at_least.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/at_least_set.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/chuffed/at_least_int.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/arg_max.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/arg_max_float.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/arg_max_int.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/arg_min.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/arg_min_float.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/arg_min_int.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/arg_sort.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/arg_sort_float.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/arg_sort_int.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/among_fn.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/among.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/alternative.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/all_equal.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/all_equal_set.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/all_equal_int.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/all_disjoint.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/alldifferent_except_0.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/all_different.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/std/all_different_set.mzn'
processing file 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/chuffed/all_different_int.mzn'
 done parsing (1.65 s)
Flattening ...
        CompilePass: Flatten with 'C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/share/minizinc/chuffed/' library ...
MIP domains ... done (0.17 s)
Optimizing ... done (0.18 s)
Converting to old FlatZinc ... done (0.19 s)
 done (0.19 s)
 done (1.85 s), max stack depth 21
Printing FlatZinc to 'dna.fzn' ... done (2.03 s)
Maximum memory 8 Mbytes.
  Flattening done, 2.04 s
      % SOLVING PHASE
FZN solver plugin, compiled  Aug 23 2018  18:30:52
Using FZN solver C:\ak\programme\MiniZinc IDE (bundled) 2.2.0/fzn-chuffed.exe for solving, parameters: -v
AAAACCCC
AAAAGGGG
AATTCCGG
AATTGGCC
AACCAACC
AACCTTGG
AACCCCAA
AACGACTG
AACGTGAC
AACGCAGT
AACGGTCA
AAGCACGT
AAGCTGCA
AAGCCATG
AAGCGTAC
ATATCGCG
ATATGCGC
ATTACGGC
ATTAGCCG
ATCCAGAG
----------
%%%mzn-stat: nodes=253
%%%mzn-stat: failures=147
%%%mzn-stat: restarts=0
%%%mzn-stat: variables=75304
%%%mzn-stat: intVars=3380
%%%mzn-stat: boolVariables=71922
%%%mzn-stat: propagators=802
%%%mzn-stat: propagations=4876
%%%mzn-stat: peakDepth=105
%%%mzn-stat: nogoods=147
%%%mzn-stat: backjumps=0
%%%mzn-stat: peakMem=0.00
%%%mzn-stat: time=0.206
%%%mzn-stat: initTime=0.188
%%%mzn-stat: solveTime=0.018
%%%mzn-stat: objective=-1
%%%mzn-stat: baseMem=0.00
%%%mzn-stat: trailMem=0.12
%%%mzn-stat: randomSeed=1535376595
   Done (overall time 2.78 s).

**Comment by guidotack** [comment]
The Chuffed version shipped with 2.2.0 was broken for Linux. We will update it in the upcoming 2.2.1 binaries.

**Comment by guidotack** [comment]
Sorry Mats: the second problem you mention is actually still present in 2.2.1. We currently cannot parse FlatZinc files correctly (because they contain flattened 1d arrays, so some constraint signatures don't match). We will release an updated MiniZinc standard library later this year that will solve this problem. For the time being you will have to run FlatZinc solvers directly (i.e., run `fzn-chuffed` rather than `minizinc --solver chuffed`).

**Comment by matsc-at-sics-se** [comment]
Guido, I'm not sure what you mean by "the second problem".
But running `fzn-chuffed` directly still doesn't seem to work: on Ubuntu:

```
matsc@teterev:~/MZN$ echo $PATH
/usr/local/bin:/home/matsc/.cabal/bin:/opt/MiniZincIDE-2.2.1-bundle-linux/bin:/opt/sicstus4.4.1/bin:/home/matsc/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
matsc@teterev:~/MZN$ minizinc --solver chuffed --fzn dna.fzn -D N=20 dna.mzn
=====ERROR=====
matsc@teterev:~/MZN$ fzn-chuffed -v dna.fzn
Segmentation fault (core dumped)
```

Do you have an ETA for a release that fixes the problem?

**Comment by guidotack** [comment]
By "the second problem" I mean the type error you get when trying to run chuffed through the `minizinc` driver.
I can confirm that I can now reproduce the segmentation fault on Ubuntu (it does not crash on macOS or Windows). This bug seems to be unrelated to another one we fixed in the 2.2.1 binaries (which caused almost all models to crash chuffed).
This is a bug in Chuffed, not in MiniZinc. It would be great it you could open a bug report on Chuffed's bug tracker (http://github.com/chuffed/chuffed/issues).

**Comment by matsc-at-sics-se** [comment]
Done: issue # 27.
